src/event_model/documents/run_start.py

```diff
@@ -25,12 +25,12 @@ class ConfigurationProjection(TypedDict):
     config_device: str
     config_index: int
     field: str
-    location: NotRequired[str]
+    location: Literal["configuration"]
```

Comment:
I'm not exactly sure what happened here; from the looks of it the JSON documents aren't changed (I would have seen it after launching the regeneration) and yet datamodel changed these `NotRequired` fields to `Literal`...

Reply:
It is more true to the pydantic model so it's better. I don't exactly know why this would change, maybe because we're specifying 3.10 as our target, if this also happens in main then it's possible it's just datamodel supporting literals now.

Either way, it's good for us, it's more accurate.

https://github.com/bluesky/event-model/blob/a896eb2145359f2b96ceca6108a143340f731af1/src/event_model/basemodels/run_start.py#L64